### PR TITLE
Head bucket should check the dns records for stat info

### PR DESCRIPTION
**Pull request by Praveenrajmani**
## Description
The Head Bucket was not querying the globalDnsConfig for the stat information

## Motivation and Context
The Head request should query the dns records to see if the bucket exists or not

## How to test this PR?
configure dns and use `bucket_exist()` in minio-py. It fails intermittently, giving the wrong information. (False for a bucket that actually exist in dns records)

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation needed
- [ ] Unit tests needed
- [ ] Functional tests needed (If yes, add [mint](https://github.com/minio/mint) PR # here: )

### CODE REVIEW

**Comment by Praveenrajmani** [comment]
Stat bucket requests always gets resolved correcly as per coreDNS

**Comment by Praveenrajmani** [comment]
Stat bucket requests always gets resolved correcly as per coreDNS

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#0ec9fe39-6f54-4105-ae3e-ea4e694eff6d) |
| mint-dist-xl.sh | :x: [more...](#90904610-c172-4a6e-91e8-daee979dc1cc) |
<a name="0ec9fe39-6f54-4105-ae3e-ea4e694eff6d"></a>
### 7954-aa5763e/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.52:31007
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 5198e506cfb5:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 3 seconds
(2/14) Running aws-sdk-java tests ... done in 2 seconds
(3/14) Running aws-sdk-php tests ... done in 48 seconds
(4/14) Running aws-sdk-ruby tests ... done in 13 seconds
(5/14) Running awscli tests ... done in 1 minutes and 29 seconds
(6/14) Running healthcheck tests ... done in 1 seconds
(7/14) Running mc tests ... FAILED in 2 minutes and 47 seconds
{
  "name": "mc",
  "duration": "124820",
  "function": "test_mirror_list_objects_storage_class",
  "status": "FAIL",
  "error": "/mint/run/core/mc/mc --config-dir /tmp/.mc-609 --quiet --no-color mirror --storage-class REDUCED_REDUNDANCY /mint/data myminio/mc-test-bucket-9790  >>> \n`/mint/data/datafile-1-MB` -> `myminio/mc-test-bucket-9790/datafile-1-MB`\n`/mint/data/datafile-0-b` -> `myminio/mc-test-bucket-9790/datafile-0-b`\n`/mint/data/datafile-1-kB` -> `myminio/mc-test-bucket-9790/datafile-1-kB`\n`/mint/data/datafile-1-b` -> `myminio/mc-test-bucket-9790/datafile-1-b`\n`/mint/data/datafile-10-MB` -> `myminio/mc-test-bucket-9790/datafile-10-MB`\n`/mint/data/datafile-1.03-MB` -> `myminio/mc-test-bucket-9790/datafile-1.03-MB`\n`/mint/data/datafile-100-kB` -> `myminio/mc-test-bucket-9790/datafile-100-kB`\n`/mint/data/datafile-11-MB` -> `myminio/mc-test-bucket-9790/datafile-11-MB`\n`/mint/data/datafile-10-kB` -> `myminio/mc-test-bucket-9790/datafile-10-kB`\n`/mint/data/datafile-33-kB` -> `myminio/mc-test-bucket-9790/datafile-33-kB`\n`/mint/data/datafile-5-MB` -> `myminio/mc-test-bucket-9790/datafile-5-MB`\n`/mint/data/datafile-129-MB` -> `myminio/mc-test-bucket-9790/datafile-129-MB`\n`/mint/data/datafile-5243880-b` -> `myminio/mc-test-bucket-9790/datafile-5243880-b`\n`/mint/data/datafile-65-MB` -> `myminio/mc-test-bucket-9790/datafile-65-MB`\n`/mint/data/datafile-6-MB` -> `myminio/mc-test-bucket-9790/datafile-6-MB`\nmc: <ERROR> Failed to copy `/mint/data/datafile-129-MB`. Please reduce your request\nmc: <ERROR> Failed to copy `/mint/data/datafile-65-MB`. Please reduce your request\nmc: <ERROR> Failed to copy `/mint/data/datafile-11-MB`. Please reduce your request\nTotal: 233.17 MiB, Transferred: 428.36 MiB, Speed: 3.43 MiB/s"
}

Executed 6 out of 14 tests successfully.
```
<a name="90904610-c172-4a6e-91e8-daee979dc1cc"></a>
### 7954-aa5763e/mint-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.53:31669
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 8ead612465d7:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 0 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 44 seconds
(4/14) Running aws-sdk-ruby tests ... done in 7 seconds
(5/14) Running awscli tests ... done in 2 minutes and 7 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 1 minutes and 13 seconds
(8/14) Running minio-dotnet tests ... done in 45 seconds
(9/14) Running minio-go tests ... done in 1 minutes and 8 seconds
(10/14) Running minio-java tests ... FAILED in 1 minutes and 27 seconds
{
  "name": "minio-java",
  "function": "listIncompleteUploads(String bucketName)",
  "duration": 700,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code=SlowDown, message=Please reduce your request, bucketName=minio-java-test-nt66re, objectName=minio-java-test-qe7beu, resource=/minio-java-test-nt66re/minio-java-test-qe7beu, requestId=15B2C97CA563245B, hostId=883f42a9-42ec-4055-84ab-9591c15b71ed)\nrequest={method=DELETE, url=http://72.28.97.53:31669/minio-java-test-nt66re/minio-java-test-qe7beu?uploadId=987ebe59-aa6e-4546-9d23-58b60e5d5a7c, headers=Host: 72.28.97.53:31669\nUser-Agent: MinIO (amd64; amd64) minio-java/dev\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20190719T105237Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20190719/us-east-1/s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=503, headers=Accept-Ranges: bytes\nContent-Length: 359\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nRetry-After: 120\nServer: MinIO/DEVELOPMENT.2019-07-19T10-32-11Z\nVary: Origin\nX-Amz-Request-Id: 15B2C97CA563245B\nX-Xss-Protection: 1; mode=block\nDate: Fri, 19 Jul 2019 10:52:38 GMT\n}\n >>> [io.minio.MinioClient.executeReq(MinioClient.java:1204), io.minio.MinioClient.execute(MinioClient.java:1070), io.minio.MinioClient.executeDelete(MinioClient.java:1347), io.minio.MinioClient.abortMultipartUpload(MinioClient.java:5149), io.minio.MinioClient.putObject(MinioClient.java:4337), io.minio.MinioClient.putObject(MinioClient.java:4183), FunctionalTest.listIncompleteUploads_test1(FunctionalTest.java:1601), FunctionalTest.runTests(FunctionalTest.java:2988), FunctionalTest.main(FunctionalTest.java:3118)]"
}

Executed 9 out of 14 tests successfully.
```

**Comment by Praveenrajmani** [comment]
Will add some changes as discussed with @vadmeste

File: cmd/bucket-handlers.go [hunk 1]
```diff
@@ -746,6 +746,18 @@ func (api objectAPIHandlers) HeadBucketHandler(w http.ResponseWriter, r *http.Re
 		return
 	}
 
+	if globalDNSConfig != nil {
```
Comment (vadmeste) [inline cmd/bucket-handlers.go]: Let's try to make etcd call only when we are not able to find the bucket in the local object storage and etcd is enabled

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#6d2586a7-0787-4cee-a03d-bd63bd8aeedc) |
| mint-dist-xl.sh | :x: [more...](#f4bb47a5-b752-450f-9c4a-2885f77ac18d) |
<a name="f4bb47a5-b752-450f-9c4a-2885f77ac18d"></a>
### 7954-aa5763e/mint-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.54:31314
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp a424714ee6cc:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 1 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 43 seconds
(4/14) Running aws-sdk-ruby tests ... done in 3 seconds
(5/14) Running awscli tests ... done in 2 minutes and 0 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 52 seconds
(8/14) Running minio-dotnet tests ... done in 33 seconds
(9/14) Running minio-go tests ... done in 1 minutes and 3 seconds
(10/14) Running minio-java tests ... FAILED in 43 seconds
{
  "name": "minio-java",
  "function": "listIncompleteUploads(String bucketName, String prefix)",
  "duration": 806,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code=SlowDown, message=Please reduce your request, bucketName=minio-java-test-q21vi4, objectName=minio-java-test-22sn3de, resource=/minio-java-test-q21vi4/minio-java-test-22sn3de, requestId=15B2CC7A18E4A1DE, hostId=37048820-8609-4bda-ab39-e515673b5cbd)\nrequest={method=DELETE, url=http://72.28.97.54:31314/minio-java-test-q21vi4/minio-java-test-22sn3de?uploadId=faaecc64-6cee-42b5-a594-e400c1422335, headers=Host: 72.28.97.54:31314\nUser-Agent: MinIO (amd64; amd64) minio-java/dev\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20190719T114725Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20190719/us-east-1/s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=503, headers=Accept-Ranges: bytes\nContent-Length: 361\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nRetry-After: 120\nServer: MinIO/DEVELOPMENT.2019-07-19T11-28-42Z\nVary: Origin\nX-Amz-Request-Id: 15B2CC7A18E4A1DE\nX-Xss-Protection: 1; mode=block\nDate: Fri, 19 Jul 2019 11:47:25 GMT\n}\n >>> [io.minio.MinioClient.executeReq(MinioClient.java:1204), io.minio.MinioClient.execute(MinioClient.java:1070), io.minio.MinioClient.executeDelete(MinioClient.java:1347), io.minio.MinioClient.abortMultipartUpload(MinioClient.java:5149), io.minio.MinioClient.putObject(MinioClient.java:4337), io.minio.MinioClient.putObject(MinioClient.java:4183), FunctionalTest.listIncompleteUploads_test2(FunctionalTest.java:1639), FunctionalTest.runTests(FunctionalTest.java:2989), FunctionalTest.main(FunctionalTest.java:3118)]"
}

Executed 9 out of 14 tests successfully.
```
<a name="6d2586a7-0787-4cee-a03d-bd63bd8aeedc"></a>
### 7954-aa5763e/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.52:31689
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp ad8b102ff68d:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 4 seconds
(2/14) Running aws-sdk-java tests ... done in 2 seconds
(3/14) Running aws-sdk-php tests ... done in 50 seconds
(4/14) Running aws-sdk-ruby tests ... done in 16 seconds
(5/14) Running awscli tests ... done in 2 minutes and 28 seconds
(6/14) Running healthcheck tests ... done in 1 seconds
(7/14) Running mc tests ... FAILED in 2 minutes and 36 seconds
{
  "name": "mc",
  "duration": "125481",
  "function": "test_mirror_list_objects_storage_class",
  "status": "FAIL",
  "error": "/mint/run/core/mc/mc --config-dir /tmp/.mc-1857 --quiet --no-color mirror --storage-class REDUCED_REDUNDANCY /mint/data myminio/mc-test-bucket-11544  >>> \n`/mint/data/datafile-1-MB` -> `myminio/mc-test-bucket-11544/datafile-1-MB`\n`/mint/data/datafile-0-b` -> `myminio/mc-test-bucket-11544/datafile-0-b`\n`/mint/data/datafile-1-b` -> `myminio/mc-test-bucket-11544/datafile-1-b`\n`/mint/data/datafile-1-kB` -> `myminio/mc-test-bucket-11544/datafile-1-kB`\n`/mint/data/datafile-10-kB` -> `myminio/mc-test-bucket-11544/datafile-10-kB`\n`/mint/data/datafile-10-MB` -> `myminio/mc-test-bucket-11544/datafile-10-MB`\n`/mint/data/datafile-1.03-MB` -> `myminio/mc-test-bucket-11544/datafile-1.03-MB`\n`/mint/data/datafile-100-kB` -> `myminio/mc-test-bucket-11544/datafile-100-kB`\n`/mint/data/datafile-11-MB` -> `myminio/mc-test-bucket-11544/datafile-11-MB`\n`/mint/data/datafile-5-MB` -> `myminio/mc-test-bucket-11544/datafile-5-MB`\n`/mint/data/datafile-129-MB` -> `myminio/mc-test-bucket-11544/datafile-129-MB`\n`/mint/data/datafile-33-kB` -> `myminio/mc-test-bucket-11544/datafile-33-kB`\n`/mint/data/datafile-65-MB` -> `myminio/mc-test-bucket-11544/datafile-65-MB`\n`/mint/data/datafile-5243880-b` -> `myminio/mc-test-bucket-11544/datafile-5243880-b`\n`/mint/data/datafile-6-MB` -> `myminio/mc-test-bucket-11544/datafile-6-MB`\nmc: <ERROR> Failed to copy `/mint/data/datafile-11-MB`. Please reduce your request\nmc: <ERROR> Failed to copy `/mint/data/datafile-10-MB`. Please reduce your request\nmc: <ERROR> Failed to copy `/mint/data/datafile-65-MB`. Please reduce your request\nmc: <ERROR> Failed to copy `/mint/data/datafile-129-MB`. Please reduce your request\nTotal: 233.17 MiB, Transferred: 504.55 MiB, Speed: 4.02 MiB/s"
}

Executed 6 out of 14 tests successfully.
```

File: cmd/bucket-handlers.go [hunk 2]
```diff
@@ -746,6 +746,18 @@ func (api objectAPIHandlers) HeadBucketHandler(w http.ResponseWriter, r *http.Re
...
+		_, err := globalDNSConfig.Get(bucket)
+		if err != nil {
+			if err == dns.ErrNoEntriesFound {
+				writeErrorResponseHeadersOnly(w, errorCodes.ToAPIErr(ErrNoSuchBucket))
+				return
+			}
+			writeErrorResponseHeadersOnly(w, toAPIError(ctx, err))
```
Comment (vadmeste) [inline cmd/bucket-handlers.go]: Also 'return' is needed after this line

File: cmd/bucket-handlers.go [hunk 3]
```diff
@@ -746,6 +746,18 @@ func (api objectAPIHandlers) HeadBucketHandler(w http.ResponseWriter, r *http.Re
...
+		}
+		writeSuccessResponseHeadersOnly(w)
```
Comment (vadmeste) [inline cmd/bucket-handlers.go]: 'return' is needed after this line

**Comment by Praveenrajmani** [comment]
PTAL @vadmeste @harshavardhana

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#bb25e887-1d10-475a-8bf8-b489c7d74d9d) |
| mint-dist-xl.sh | :x: [more...](#253245cd-5e35-493b-b80f-1d6ec2c1e0af) |
<a name="bb25e887-1d10-475a-8bf8-b489c7d74d9d"></a>
### 7954-d0f4c17/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.60:30397
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 1617ce1e8115:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 3 seconds
(2/14) Running aws-sdk-java tests ... done in 2 seconds
(3/14) Running aws-sdk-php tests ... done in 48 seconds
(4/14) Running aws-sdk-ruby tests ... done in 13 seconds
(5/14) Running awscli tests ... done in 2 minutes and 25 seconds
(6/14) Running healthcheck tests ... done in 1 seconds
(7/14) Running mc tests ... FAILED in 2 minutes and 50 seconds
{
  "name": "mc",
  "duration": "124415",
  "function": "test_mirror_list_objects_storage_class",
  "status": "FAIL",
  "error": "/mint/run/core/mc/mc --config-dir /tmp/.mc-7666 --quiet --no-color mirror --storage-class REDUCED_REDUNDANCY /mint/data myminio/mc-test-bucket-7400  >>> \n`/mint/data/datafile-1-MB` -> `myminio/mc-test-bucket-7400/datafile-1-MB`\n`/mint/data/datafile-0-b` -> `myminio/mc-test-bucket-7400/datafile-0-b`\n`/mint/data/datafile-1-kB` -> `myminio/mc-test-bucket-7400/datafile-1-kB`\n`/mint/data/datafile-1-b` -> `myminio/mc-test-bucket-7400/datafile-1-b`\n`/mint/data/datafile-10-MB` -> `myminio/mc-test-bucket-7400/datafile-10-MB`\n`/mint/data/datafile-1.03-MB` -> `myminio/mc-test-bucket-7400/datafile-1.03-MB`\n`/mint/data/datafile-100-kB` -> `myminio/mc-test-bucket-7400/datafile-100-kB`\n`/mint/data/datafile-10-kB` -> `myminio/mc-test-bucket-7400/datafile-10-kB`\n`/mint/data/datafile-11-MB` -> `myminio/mc-test-bucket-7400/datafile-11-MB`\n`/mint/data/datafile-33-kB` -> `myminio/mc-test-bucket-7400/datafile-33-kB`\n`/mint/data/datafile-129-MB` -> `myminio/mc-test-bucket-7400/datafile-129-MB`\n`/mint/data/datafile-5-MB` -> `myminio/mc-test-bucket-7400/datafile-5-MB`\n`/mint/data/datafile-6-MB` -> `myminio/mc-test-bucket-7400/datafile-6-MB`\n`/mint/data/datafile-5243880-b` -> `myminio/mc-test-bucket-7400/datafile-5243880-b`\n`/mint/data/datafile-65-MB` -> `myminio/mc-test-bucket-7400/datafile-65-MB`\nmc: <ERROR> Failed to copy `/mint/data/datafile-10-MB`. Please reduce your request\nmc: <ERROR> Failed to copy `/mint/data/datafile-65-MB`. Please reduce your request\nmc: <ERROR> Failed to copy `/mint/data/datafile-129-MB`. Please reduce your request\nmc: <ERROR> Failed to copy `/mint/data/datafile-11-MB`. Please reduce your request\nTotal: 233.17 MiB, Transferred: 472.67 MiB, Speed: 3.80 MiB/s"
}

Executed 6 out of 14 tests successfully.
```
<a name="253245cd-5e35-493b-b80f-1d6ec2c1e0af"></a>
### 7954-d0f4c17/mint-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.61:31073
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp b77f6bd86ccb:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 1 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 45 seconds
(4/14) Running aws-sdk-ruby tests ... done in 3 seconds
(5/14) Running awscli tests ... done in 1 minutes and 17 seconds
(6/14) Running healthcheck tests ... done in 1 seconds
(7/14) Running mc tests ... done in 59 seconds
(8/14) Running minio-dotnet tests ... done in 57 seconds
(9/14) Running minio-go tests ... done in 1 minutes and 10 seconds
(10/14) Running minio-java tests ... FAILED in 51 seconds
{
  "name": "minio-java",
  "function": "removeIncompleteUpload(String bucketName, String objectName)",
  "duration": 688,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code=SlowDown, message=Please reduce your request, bucketName=minio-java-test-3ve9f, objectName=minio-java-test-38pkj6p, resource=/minio-java-test-3ve9f/minio-java-test-38pkj6p, requestId=15B2D04C24879C4E, hostId=5a5176c2-5fa7-4765-8d9c-fab873e99029)\nrequest={method=DELETE, url=http://72.28.97.61:31073/minio-java-test-3ve9f/minio-java-test-38pkj6p?uploadId=b201f5ab-eb52-433e-acb1-8f6c7b047e14, headers=Host: 72.28.97.61:31073\nUser-Agent: MinIO (amd64; amd64) minio-java/dev\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20190719T125726Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20190719/us-east-1/s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=503, headers=Accept-Ranges: bytes\nContent-Length: 359\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nRetry-After: 120\nServer: MinIO/DEVELOPMENT.2019-07-19T12-38-24Z\nVary: Origin\nX-Amz-Request-Id: 15B2D04C24879C4E\nX-Xss-Protection: 1; mode=block\nDate: Fri, 19 Jul 2019 12:57:26 GMT\n}\n >>> [io.minio.MinioClient.executeReq(MinioClient.java:1204), io.minio.MinioClient.execute(MinioClient.java:1070), io.minio.MinioClient.executeDelete(MinioClient.java:1347), io.minio.MinioClient.abortMultipartUpload(MinioClient.java:5149), io.minio.MinioClient.putObject(MinioClient.java:4337), io.minio.MinioClient.putObject(MinioClient.java:4183), FunctionalTest.removeIncompleteUploads_test(FunctionalTest.java:1719), FunctionalTest.runTests(FunctionalTest.java:2992), FunctionalTest.main(FunctionalTest.java:3118)]"
}

Executed 9 out of 14 tests successfully.
```

**Comment by vadmeste** [comment]
As discussed with @Praveenrajmani, S3 clients are supposed to make virtual host requests with HEAD bucket, minio-go has it, but it seems minio-py doesn't have it.

But I think is also good to have this PR

**Comment by harshavardhana** [comment]
This is not a correct PR, bucketForwardHandler already handles the required checks. You won't even reach HeadBucket without going through that.

We should close this PR.

**Comment by harshavardhana** [comment]
The correct fix was merged #7957 which has fixed the customer issue. Will close this PR.

**Pull request closed without merging**